interview-questions.md

```diff
@@ -4,55 +4,58 @@ Answer the following questions. First, without external resources. Challenge you
 
 1. What is a function? Why would you use one?
 
-  Your answer:
+  Your answer: a function is a dynamic variable. It is set up to take in information and process it according to design
 
-  Researched answer:
+  Researched answer: In programming, a named section of a program that performs a specific task. In this sense, a function is a type of procedure, routine or machine. Some programming languages make a distinction between a function, which returns a value, and a procedure, which performs some operation but does not return a value. (google- augmented by me)
 
 
 
 2. What is the difference between map and filter?
 
-  Your answer:
+  Your answer: Map can run through an array and return with a new array giving us useful information where as a filter can create a new array with desired data
```

Comment:
great! i didn't see a researched answer for this question. you may want to add a bit more to this.